music21/roman.py

```diff
@@ -748,83 +690,66 @@ def romanNumeralFromChord(chordObj,
+    Music21 generally handles this with the :class:`music21.roman.Minor67Default`
+    as explained in the :class:`music21.roman.romanNumeral` and especially the
+    :meth:`~music21.roman.RomanNumeral.adjustMinorVIandVIIByQuality`.
 
+    This function uses a variant on the same approach,
+    but robust handling is a work in progress.
+    Here are some examples of behaviour as currently expected:
 
-    Former bugs:
-
-    Should be iii7
-
-    >>> romanNumeral11 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['E4', 'G4', 'B4', 'D5']),
-    ...     key.Key('C'),
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['A4', 'C5', 'E-5']),
+    ...     key.Key('c'),
     ...     )
-    >>> romanNumeral11
-    <music21.roman.RomanNumeral iii7 in C major>
-
-    Should be viø7  # gave vio7
-
-    >>> roman.romanNumeralFromChord(chord.Chord('A3 C4 E-4 G4'), key.Key('c'))
-    <music21.roman.RomanNumeral viø7 in c minor>
-
-    Should be viiø7  # gave viio7
-
-    >>> roman.romanNumeralFromChord(chord.Chord('A3 C4 E-4 G4'), key.Key('B-'))
-    <music21.roman.RomanNumeral viiø7 in B- major>
-
-
-    Should be I#853
+    <music21.roman.RomanNumeral vio in c minor>
 
-    >>> romanNumeral9 = roman.romanNumeralFromChord(
-    ...     chord.Chord(['C4', 'E5', 'G5', 'C#6']),
-    ...     key.Key('C'),
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['A-4', 'C5', 'E-5']),
+    ...     key.Key('c'),
     ...     )
-    >>> romanNumeral9
-    <music21.roman.RomanNumeral I#853 in C major>
-
-
-    Not an augmented 6th:
+    <music21.roman.RomanNumeral bVI in c minor>
 
     >>> roman.romanNumeralFromChord(
-    ...     chord.Chord('E4 G4 B-4 C#5')
+    ...     chord.Chord(['B4', 'D5', 'F5']),
+    ...     key.Key('c'),
     ...     )
-    <music21.roman.RomanNumeral io6b5b3 in c# minor>
-
-
-    OMIT_FROM_DOCS
-
-
-    Note that this should be III+642 gives III+#642 (# before 6 is unnecessary)
-
-    # >>> roman.romanNumeralFromChord(chord.Chord('B3 D3 E-3 G3'), key.Key('c'))
-    # <music21.roman.RomanNumeral III+642 in c minor>
-
+    <music21.roman.RomanNumeral viio in c minor>
 
-    These two are debatable -- is the harmonic minor or the natural minor used as the basis?
+    >>> roman.romanNumeralFromChord(
+    ...     chord.Chord(['B-4', 'D5', 'F5']),
+    ...     key.Key('c'),
+    ...     )
+    <music21.roman.RomanNumeral bVII in c minor>
 
-    # >>> roman.romanNumeralFromChord(chord.Chord('F4 A4 C5 E-5'), key.Key('c'))
-    # <music21.roman.RomanNumeral IVb753 in c minor>
-    # <music21.roman.RomanNumeral IV75#3 in c minor>
+    Finally, diminished and half-diminished seventh chords can
+    omit the third and still be diminished
+    (note also from this example that chords can be created from a string):
 
-    # >>> roman.romanNumeralFromChord(chord.Chord('F4 A4 C5 E5'), key.Key('c'))
-    # <music21.roman.RomanNumeral IV7 in c minor>
-    # <music21.roman.RomanNumeral IV#75#3 in c minor>
```

Comment:
Same

Reply:
Same here -- do not remove tests that show bugs.